src/HttpClientInterception/HttpRequestInterceptionBuilder.cs

```diff
@@ -111,26 +111,46 @@ public HttpRequestInterceptionBuilder For(Predicate<HttpRequestMessage>? predica
     /// Pass a value of <see langword="null"/> to remove a previously-registered custom request matching predicate.
     /// </remarks>
     public HttpRequestInterceptionBuilder For(params Predicate<HttpRequestMessage>[] predicates)
+    {
+        return For(predicates
+            .Select(predicate => new Func<HttpRequestMessage, Task<bool>>((message) => Task.FromResult(predicate(message))))
+            .ToArray());
+    }
+
+    /// <summary>
+    /// Configures the builder to match any request that meets all the criteria defined by the specified predicates.
+    /// </summary>
+    /// <param name="predicates">
+    /// Two or more delegates to a method which returns <see langword="true"/> if the
+    /// request is considered a match; otherwise <see langword="false"/>.
+    /// </param>
+    /// <returns>
+    /// The current <see cref="HttpRequestInterceptionBuilder"/>.
+    /// </returns>
+    /// <remarks>
+    /// Pass a value of <see langword="null"/> to remove a previously-registered custom request matching predicate.
+    /// </remarks>
+    public HttpRequestInterceptionBuilder For(params Func<HttpRequestMessage, Task<bool>>[] predicates)
     {
         if (predicates?.Length == 1)
         {
             return For(predicates[0]);
         }
 
-        _requestMatcher = predicates != null ?
-        (p) =>
-        {
-            foreach (var predicate in predicates)
+        _requestMatcher = predicates == null || predicates.Length == 0
+            ? null
+            : async (p) =>
             {
-                if (!predicate(p))
+                foreach (var predicate in predicates)
                 {
-                    return Task.FromResult(false);
+                    if (!await predicate(p).ConfigureAwait(false))
```

Comment:
The comment on the other method reminds me that these delegates need to be checked for null before use. Best doing it up front once, rather than every time this lambda is invoked.